src/common/mfu_io.c

```diff
@@ -326,6 +697,26 @@ int mfu_close(const char* file, int fd)
     return rc;
 }
 
+int mfu_file_close(const char* file, mfu_file_t* mfu_file)
+{
+    if (mfu_file->type == POSIX) {
+        int rc = mfu_close(file, mfu_file->fd);
+        return rc;
+    } else if (mfu_file->type == DAOS) {
+        int rc = daos_close(file, mfu_file);
+        return rc;
+    } else {
+        MFU_ABORT(-1, "File type not known: %s errno=%d (%s)",
+                  file, errno, strerror(errno));
+    }
+}
+
+off_t daos_lseek(const char* file, mfu_file_t* mfu_file, off_t pos, int whence)
+{
+    /* noop becuase daos have an lseek */
+    return 0;
+}
+
```

Comment:
Let's have this set the mfu_file->offset value based on pos/whence values.  For this PR, you could just worry about SEEK_SET and abort if whence is not SEEK_SET.

Reply:
@adammoody ok, this is what I updated to:

```C
    if (whence == SEEK_SET) {
        mfu_file->offset = (daos_off_t)pos;
    } else {
        MFU_ABORT(-1, "whence type not known: %d errno=%d (%s)",
                  whence, errno, strerror(errno));
    }
```